src/main/java/mmosii/bookstore/service/impl/BookServiceImpl.java

```diff
@@ -63,9 +70,23 @@ public List<BookDto> search(BookSearchParametersDto searchParameters) {
                 .toList();
     }
 
+    @Override
+    public List<BookDtoWithoutCategoryIds> findBooksByCategory(Pageable pageable, Long id) {
+        return bookRepository.findAllByCategoryId(id).stream()
+                .map(bookMapper::toDtoWithoutCategories)
+                .toList();
+    }
+
     private void checkIfBookExistsById(Long id) {
         if (!bookRepository.existsById(id)) {
             throw new EntityNotFoundException("Can't find book with id " + id);
         }
     }
+
+    private void setCategories(CreateBookRequestDto requestDto, Book book) {
+        if (requestDto.categoryIds() != null) {
+            book.setCategories(new HashSet<>(
+                    categoryRepository.findAllById(requestDto.categoryIds())));
```

Comment:
I would suggest using `getReference` since you don't need to load the caregories here - you just need to create a mapping